### PR TITLE
chore: replace `react-native-safe-area-context`

### PR DESCRIPTION
**Pull request by tido64**
### Description

Replace `react-native-safe-area-context` with a module we have better control over.

**Blocked by:**
- [x] https://github.com/microsoft/rnx-kit/pull/2942
- [x] https://github.com/microsoft/rnx-kit/pull/2946
- [x] https://github.com/microsoft/rnx-kit/pull/2954

### Platforms affected

- [ ] Android
- [ ] iOS
- [ ] macOS
- [ ] Windows

### Test plan

Build the example app (any platform will do).

| Android | iOS | macOS | Windows |
| :-: | :-: | :-: | :-: |
| Screenshot_1706795556 | image | image | image |

### CODE REVIEW

**Comment by kelset** [comment]
tested locally as per https://github.com/microsoft/rnx-kit/pull/2942#issuecomment-1927617197

once we open this one up properly and conflicts are addressed I can give it another pass, overall LGTM

File: example/App.tsx [hunk 1]
```diff
@@ -98,6 +98,17 @@ function useIsFabricComponent() {
...
+function useLocalStorageStatus() {
+  const [localValue, setLocalValue] = useState("Checking");
+  useEffect(() => {
+    const key = "sample/local-storage";
+    window?.localStorage?.setItem(key, "Available");
+    setLocalValue(window?.localStorage?.getItem(key) ?? "Error");
+    return () => window?.localStorage?.removeItem(key);
+  }, []);
+  return localValue;
+}
```
Comment (kelset) [inline example/App.tsx:110]: ngl it kind of got shivers down my spine for a second seeing this web code - damn 🔥

**Comment by kelset** [comment]
Local testing:





| Android (old arch) | Android (new arch)  | Android (new arch + bridgeless)  | iOS (old arch) | iOS (new arch + bridgeless) |
| :-: | :-: | :-: | :-: | :-: |
| Screenshot_1707230631 | Screenshot_1707230986  |  Screenshot_1707231387 | Simulator Screenshot - iPhone 15 - 2024-02-06 at 14 35 17 | simulator_screenshot_7B7C54C5-7542-4B55-B369-C714EA7B4FE5 |

would be good to investigate why with bridgeless it errors out but then it works (see the screenshot in the table, after hitting dismiss). Let's fix it in a follow up:

Screenshot_1707230795

**Review by kelset — APPROVED** [review]
let's go ahead and address the android bridgeless error separately

**Comment by tido64** [comment]
FYI, it looks like the `NullPointerException` exception has been addressed: https://github.com/facebook/react-native/commit/68bc219266517af09c6f4ef106b25ec0eea37e06

**Comment by cortinico** [comment]
> would be good to investigate why with bridgeless it errors out but then it works (see the screenshot in the table, after hitting dismiss). Let's fix it in a follow up:

This is fixed in 0.73.4 (see relevant commit here https://github.com/reactwg/react-native-releases/discussions/101#discussioncomment-8307715)

**Comment by tido64** [comment]
> This is fixed in 0.73.4 (see relevant commit here [reactwg/react-native-releases#101 (comment)](https://github.com/reactwg/react-native-releases/discussions/101#discussioncomment-8307715))

Thanks for the context

**Pull request merged**